cell_workers/pokemon_catch_worker.py

```diff
@@ -105,6 +106,19 @@ def work(self):
                                 if status is 3:
                                     print('[x] Oh no! ' + str(pokemon_name) + ' vanished! :(')
                                 if status is 1:
+                                    self.api.get_inventory()
+                                    response_dict = self.api.call()
+                                    if self.config.smart_catch:
+                                        id_cp_tuples = self.get_id_cp_tuples_for_pokemonid(pokemon['pokemon_data']['pokemon_id'],response_dict)
+                                        prev_id, prev_cp = (0,0)
+                                        for id_cp in id_cp_tuples:
+                                            next_id,next_cp = id_cp
+                                            if next_cp < prev_cp:
+                                                self.transfer_pokemon(next_id)
+                                            elif prev_id != 0:
+                                                self.transfer_pokemon(prev_id)
+                                            prev_id,prev_cp = id_cp
+
                                     if cp < self.config.cp:
```

Comment:
@mohitmun I still see a possibility here of code getting inside `smart_catch` if and inside this one. So I have a new idea! Instead of adding a new option for smart catching, why not use `-c` for this? You can make `-c` accept a string and then you check if the string is `smart` to use smart catch. If the string isn't `smart`, then you try to convert it to int.